content/blog/en/building-dojo-app.md

```diff
@@ -0,0 +1,1124 @@
+					home: <Home />,
+					menu: <DrinkList drinks={drinks} />
+				}}
+			</Outlet>
+		</div>
+	);
+});
+```
+
+The store middleware is used to get access to the drinks from our external source. The first time the widget renders, drinks will be `undefined` so we can use the `executor` of the store to execute our `fetchDrinks` process.
+
+```ts
+// use the store to get state
+const { executor, get, path } = store;
+const drinks = get(path('drinks'));
+
+if (!drinks) {
+    executor(fetchDrinks)({});
+}
+```
+
+When the process executes, it will kick of another render of the widget when the `drinks` array is updated. The `drinks` are then passed to our `DrinkList`, via `<DrinkList drinks={drinks} />`. So let's take a look at that next.
+
+## Keep a list
+
+The `DrinkList` is going to represent our menu page. It will display an image, name, and price for the drinks our customers can choose from. This is accomplished with a list of `<DrinkCard>` widgets.
+
+```tsx
+// src/widgets/DrinkList.tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+import DrinkCard from './DrinkCard';
+
+import { State } from '../interfaces';
+
+import * as css from './styles/DrinkList.m.css';
+
+const factory = create().properties<State>();
+
+export default factory(function DrinkList({ properties }) {
+	const { drinks = [] } = properties();
+	const elems = drinks.map((drink) => <DrinkCard {...drink} />);
+
+	return <div classes={[css.root]}>{elems}</div>;
+});
+```
+
+The `DrinkCard` widget is going to provide a nice preview for our stellar coffee drinks!
+
+```tsx
+// src/widgets/DrinkCard.tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+import Avatar from '@dojo/widgets/avatar';
+import Card from '@dojo/widgets/card';
+import Label from '@dojo/widgets/label';
+import Link from '@dojo/framework/routing/Link';
+
+import { Drink } from '../interfaces';
+import * as css from './styles/DrinkCard.m.css';
+
+const factory = create().properties<Drink>();
+
+export default factory(function DrinkCard({ properties }) {
+	const { id, name = 'Coffee', price, imageUrl: image } = properties();
+
+	return (
+		<div classes={[css.root]}>
+			<Card title={name}>
+				{{
+					content: (
+						<div classes={[css.content]}>
+							<Avatar
+								variant="circle"
+								size="large"
+								src={image}
+							></Avatar>
+							<Label>${price.toFixed(2)}</Label>
+						</div>
+					),
+					actionButtons: (
+						<Link
+							to="drink"
+							params={{
+								id: `${String(id)}`,
+							}}
+						>
+							Add
+						</Link>
+					),
+				}}
+			</Card>
+		</div>
+	);
+});
+```
+
+The `DrinkCard` utilizes some widgets from `@dojo/widgets`. We're going to use the `Card` to wrap the widget. We'll be using the `Avatar` to display a circular shaped image for us. We can then use the `Link` widget from `@dojo/framework/routing/Link` to create an Add button. This will let us route to a specific drink page to customize the drink and add it to our order.
```

Comment:
```suggestion
The `DrinkCard` utilizes some widgets from `@dojo/widgets`. We're going to use the `Card` to wrap the widget. We'll be using the `Avatar` to display a circular shaped image. We can then use the `Link` widget from `@dojo/framework/routing/Link` to create an Add button. This will let us route to a specific drink page to customize the drink and add it to our order.
```